src/components/shared/ThemeSwitch.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useState } from 'react'
+import { Switch, HStack, Text, useColorMode } from 'native-base'
+import asyncStorageService from 'services/AsyncStorageService'
+
+type ThemeSwitchProps = {
+  text: string
+}
+
+export const ThemeSwitch = ({ text }: ThemeSwitchProps) => {
+  const { colorMode, toggleColorMode } = useColorMode()
+
+  const [isChecked, setisChecked] = useState<boolean>(false)
+  const toggleDarkMode = async () => {
+    toggleColorMode()
+    const darkMode = isChecked ? false : true
+    setisChecked(darkMode)
+    await asyncStorageService.setItem('darkMode', darkMode)
+  }
+  return (
+    <HStack space={2} justifyContent="space-between">
+      <Text>{text}</Text>
+      <Switch
+        onToggle={() => {
+          toggleDarkMode()
+        }}
```

Comment:
```suggestion
        onToggle={toggleDarkMode}
```

Reply:
Tried this method, got a loop re-render, modified the first method so its a bit better then before